polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -657,8 +689,14 @@ fn handle_parent_process(
 		status,
 	);
 
+	#[cfg(not(feature = "x-shadow"))]
 	let usage_after = nix::sys::resource::getrusage(UsageWho::RUSAGE_CHILDREN)
 		.map_err(|errno| error_from_errno("getrusage after", errno))?;
+	#[cfg(feature = "x-shadow")]
+	// The getrusage system call is not supported under Shadow simulation.
+	// As a workaround, we return a zeroed structure. This is safe, but
+	// it makes CPU usage calculations meaningless.
+	let usage_after = unsafe { std::mem::zeroed() };
```

Comment:
In general, but as an example here, it would be nice to isolate these differences as much as possible.

E.g. provide a wrapper function to retrieve the data we actually care about and do the #[cfg...) dance inside. Best case this can then be used even for further platform support.